File: JobDescription/src/src/validator/TaskValidator.java
package validator;

import domain.Task;
import exception.TaskException;

/**
 * Created by claudiu on 12.10.2016.
 */
public class TaskValidator implements Validator<Task, TaskException> {
    public void validate(Task task) throws TaskException {
        if (task.getDescription().isEmpty()) {
            throw new TaskException("Description can't be empty!");
        }
    }
}

File: JobDescriptionCsharp/JobDescriptionCsharp/validator/TaskValidator.cs
using JobDescriptionCsharp.domain;
using JobDescriptionCsharp.exception;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace JobDescriptionCsharp.validator
{
    public class TaskValidator : IValidator<Task>
    {
        public void validate(Task task)
        {
            if (task.description == "")
            {
                throw new TaskException("The description of task can't be empty!");
            }
        }
    }
}
